### ISSUE TITLE
Custom delivery methods

### ISSUE THREAD

**Issue by excid3** [excid3/noticed#2]
Currently, we have the ability to specify delivery methods by name, but it would be nice to be able to specify them by providing the class.

For example:
```ruby
  deliver_by :database, format: :attributes_for_database
  deliver_by :websocket
  deliver_by :email, mailer: UserMailer
  
  deliver_by DiscordNotification
```

We could then define a `DiscordNotification` class and it would be instantiated automatically for the delivery.

This probably is close to working. The `deliver_by` method on `Noticed::Base` and the `perform` method are the two main things we'll need to explore.

**Comment by excid3** [comment]
Also need to make sure this is throughly documented.

Basically the only thing a custom delivery method needs to do is:
1. Inherit from `Noticed::DeliveryMethods::Base`
2. Override the `deliver()` method

**Comment by BilalBudhani** [comment]
I will set up a test app next week and get myself familiar with the gem. I think then I will be able to provide better feedback on this.

**Comment by excid3** [comment]
I believe I got this working as well yesterday. The delivery methods must be passed in as strings though due to a limitation of ActiveJob serializers in Rails 6.0 but is fixed in 6.1.

For now I changed the syntax to be the following:

```ruby
  deliver_by :database, format: :attributes_for_database
  deliver_by :websocket
  deliver_by :email, mailer: "UserMailer"

  deliver_by :discord, class: "DiscordNotification"
```

It should accept a `deliver_by DiscordNotification`, but I'm not testing against Rails 6.1, so that may need to change in the future.
